shapes/neurosciencegraph/datashapes/core/workflowengine/schema.json

```diff
@@ -0,0 +1,76 @@
+{
+  "@context": [
+    "https://incf.github.io/neuroshapes/contexts/schema.json",
+    {
+      "this": "https://neuroshapes.org/dash/workflowengine/shapes/"
+    }
+  ],
+  "@id": "https://neuroshapes.org/dash/workflowengine",
+  "@type": "nxv:Schema",
+  "imports": [
+    "https://neuroshapes.org/dash/softwareagent"
+  ],
+  "shapes": [
+    {
+      "@id": "this:WorkflowEngineShape",
+      "@type": "sh:NodeShape",
+      "description": "Software agent which executes automated workflows.",
+      "targetClass": "nsg:WorkflowEngine",
+      "nodeKind": "sh:BlankNodeOrIRI",
+      "and": [
+        {
+          "node": "https://neuroshapes.org/dash/softwareagent/shapes/SoftwareAgentShape"
+        },
+        {
+          "property": [
+            {
+              "path": "schema:name",
+              "name": "Name.",
+              "description": "Name of the workflow that engine was running.",
+              "datatype": "xsd:string",
+              "minCount": 1,
+              "maxCount": 1
+            },
+            {
+              "path": "schema:description",
+              "name": "Description.",
+              "description": "Optional description of the workflow.",
+              "datatype": "xsd:string",
+              "maxCount": 1
+            },
+            {
+              "path": "schema:version",
+              "name": "Version.",
+              "description": "Version of the workflow engine.",
+              "datatype": "xsd:string",
+              "minCount": 1,
+              "maxCount": 1
+            },
+            {
+              "path": "nsg:parameter",
+              "name": "Parameter.",
+              "description": "Parameters which were used to run the workflow.",
+              "datatype": "xsd:string",
+              "minCount": 1,
+              "maxCount": 1
+            },
+            {
```

Comment:
I recommend to have every file (e.g. config) in a separate entity so that the file can be described  referenced later on. I suggest the following:

```
{
      "path": "nsg:configuration",
      "name": "Workflow Engine Configuration file.",
      "description": "Workflow configuration file.",
      "class": "nsg:Configuration",
      "node":"https://neuroshapes.org/commons/entity/shapes/EntityShape",
      "maxCount": 1
}
```

Reply:
But the files are separate entities in v1, they are not attachments. Am I missing something?

Reply:
They are separate entities indeed, hence my proposal. The shape above required the configuration file to be a separate entity with type nsg:Configuration and with a distribution property.

Files should have identifier (URIs) and not blank nodes.

Reply:
I just don't get it what value brings hiding one URIs with the others. DataDownload contentUrl will point to a file nexus entity. If we wrap that in nsg:Configuration it will just add one hop through the graph without much of additional semantics IMHO, especially if we wrap it in very generic prov:Entity.

Reply:
We need to stick to a common pattern when it come to record and get the downloadable form of an entity across Neuroshapes.

So far:

* A piece of data (e.g. file - config or not) should be an entity and identified by a URI so that it can be referenced in many contexts (e.g within a workflow engine that uses it as config file  but also to describe the file provenance).
* Given an entity (e.g. a file), the way to get the corresponding data is through a distribution property value of type schema:DataDownload with contentURL (to directly download the data) or url (to ). The presence of the distribution property is what allow consistency in term of data access across Neuroshapes.

Reply:
I understand tough your point that it adds an extra hop when it come to navigate the graph. So it is shorter path (-1) from the entity to the download url but only in the simulation domain VS same path length from the entity to the download url but across all Neuroshapes.

Reply:
Done